src/controller/artifact/processor/default.go

```diff
@@ -33,9 +43,10 @@ var (
 // currently, it only tries to parse the artifact type from media type
 type defaultProcessor struct {
 	mediaType string
```

Comment:
The property `mediaType` isn't needed anymore as we passing `*artifact.Artifact` as the parameter of functions and it contains the `MediaType` already